Mathlib/Condensed/Light/CartesianClosed.lean

```diff
@@ -0,0 +1,33 @@
+/-
+Copyright (c) 2024 Dagur Asgeirsson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Dagur Asgeirsson
+-/
+import Mathlib.Condensed.Light.Limits
+import Mathlib.CategoryTheory.Closed.Types
+import Mathlib.CategoryTheory.Sites.CartesianClosed
+/-!
+
+# Light condensed sets form a cartesian closed category
+-/
+
+
+universe u
+
+noncomputable section
+
+open CategoryTheory
+
+variable {C : Type u} [SmallCategory C]
+
+instance : CartesianClosed (LightProfinite.{u}ᵒᵖ ⥤ Type u) := by
+  -- We need to consider `LightProfinite` as a small category:
+  let e : LightProfinite.{u}ᵒᵖ ⥤ Type u ≌ (SmallModel.{u, u, u+1} LightProfinite.{u})ᵒᵖ ⥤ Type u :=
+    Functor.asEquivalence ((whiskeringLeft _ _ _).obj (equivSmallModel _).op.inverse)
+  -- Now we conclude because the category of functors from a small category to a cartesian closed
+  -- category is cartesian closed.
+  exact cartesianClosedOfEquiv e.symm
```

Comment:
It seems to me this may intersect slightly with #13841 or #13710.

Also, it would be good to show the existence of suitable adjoint functors (the result ` Presheaf.isLeftAdjoint_of_preservesColimits` used in `CategoryTheory.Closed.Types`) under more general assumptions, so that it may be applied to condensed or light condensed sets.

Reply:
I wasn't aware of these two PRs. I'm not sure exactly what you mean by generalizing the result `Presheaf.isLeftAdjoint_of_preservesColimits`, but it seems to me that #13710 is proving that `C ⥤ Type max w v u` is cartesian closed, which can be obtained by generalizing what I do for `CompHaus` as follows:

```lean
instance {C : Type u} [Category.{v} C] : CartesianClosed (C ⥤ Type (max u v w)) :=
  let e : C ⥤ Type (max u v w) ≌
    (ULiftHom.{max u v w} (ULift.{max u v w} C)) ⥤ Type (max u v w) :=
      Functor.asEquivalence ((whiskeringLeft _ _ _).obj
        (ULift.equivalence.trans ULiftHom.equiv).inverse)
  cartesianClosedOfEquiv e.symm
```

It might be useful to have the more explicit approach of #13710, I don't know.

Reply:
Oh I see, #13710 has something to do with `ChosenFiniteProducts`, so it's a bit different, and I guess the point there is to be explicit somehow.

Reply:
You proof uses `Presheaf.isLeftAdjoint_of_preservesColimits` (though not directly). I feel it would be cleaner if a more general (universe-wise) version of this was used more directly here instead of transferring the `CartesianClosed` instances (which presumably gives no control or understanding on the right adjoint functors that are part of the data).

Reply:
I see your point, but I think in most cases one doesn't need any "control or understanding" of the right adjoints other than the fact that they are right adjoints. Similarly to that one doesn't usually need any definitional properties of limits/colimits other than the formal ones, etc...

But I'll see if I can generalize universes in `Presheaf.isLeftAdjoint_of_preservesColimits` at some point, I won't have that much Lean time in the coming weeks though.